hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -4047,6 +4047,14 @@ private void chooseExcessRedundancyStriped(BlockCollection bc,
         List<DatanodeStorageInfo> replicasToDelete = placementPolicy
             .chooseReplicasToDelete(nonExcess, candidates, (short) 1,
                 excessTypes, null, null);
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Choose redundant EC replicas to delete from blk_{} which is located in {}",
+              sblk.getBlockId(), storage2index);
+          LOG.debug("Storages with candidate blocks to be deleted: {}", candidates);
+          LOG.debug("Storages with blocks to be deleted: {}", replicasToDelete);
+        }
+        Preconditions.checkArgument(candidates.containsAll(replicasToDelete),
+            "The EC replicas to be deleted is not in the candidate list");
```

Comment:
@tomscut Thanks for updating it. There seems to be a grammatical error. The others look good to me.
```suggestion
            "The EC replicas to be deleted are not in the candidate list");
```

Reply:
I am sorry. And I updated it. Thanks.